// webpack-init/lib/loaderConfig.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = loaderConfig;
function loaderConfig(includePath) {
  var babelConfig = {
    test: /\.jsx?/,
    loaders: ['babel'],
    include: includePath,
    modules: ['babel-loader', 'babel-core', 'babel-preset-es2015'],
    query: {
      presets: ['es2015']
    }
  };

  var postCSSLoaderConfig = {
    test: /\.css$/,
    loaders: ['style', 'css', 'postcss'],
    modules: ['postcss-loader']
  };

  var lessLoaderConfig = {
    test: /\.less$/,
    loaders: ['style', 'css', 'less'],
    modules: ['less-loader']
  };

  var sassLoaderConfig = {
    test: /\.scss$/,
    loaders: ['style', 'css', 'sass'],
    modules: ['sass-loader', 'node-sass']
  };

  var urlLoaderConfig = {
    test: /\.(png|jpg)$/,
    loaders: ['url'],
    query: {
      limit: 8192
    },
    modules: ['url-loader']
  };

  return {
    babel: babelConfig,
    postcss: postCSSLoaderConfig,
    less: lessLoaderConfig,
    sass: sassLoaderConfig,
    url: urlLoaderConfig
  };
}
