// webpack-init/lib/promptInputs.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var base = exports.base = [{
  type: 'input',
  name: 'configFileName',
  message: 'Enter the file name for your webpack config',
  default: function _default() {
    return 'webpack.config.js';
  }
}, {
  type: 'input',
  name: 'entry',
  message: 'Enter the filename for your entry',
  default: function _default() {
    return './index.js';
  }
}, {
  type: 'input',
  name: 'outputPath',
  message: 'Enter the output path, this is where your images and js will be output at',
  default: function _default() {
    return './';
  }
}, {
  type: 'input',
  name: 'outputFilename',
  message: 'Enter the output filename',
  default: function _default() {
    return 'bundle.js';
  }
}, {
  type: 'confirm',
  name: 'publicPathRequired',
  message: 'Is your output in a different directory?'
}, {
  type: 'input',
  name: 'outputPublicPath',
  message: 'Enter the path to where you will point to in your index.html e.g. /static/',
  when: function when(_ref) {
    var publicPathRequired = _ref.publicPathRequired;
    return publicPathRequired;
  }
}, {
  type: 'checkbox',
  name: 'loaders',
  message: 'What loaders would you like to include?\n              Note some loaders may require extra configuration. See\n              https://webpack.github.io/docs/list-of-loaders.html',
  choices: [' babel', ' postcss', ' sass', ' less', ' url']
}];
